pkg/plugin/runtime_subprocess_getter.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright The Helm Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugin
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"slices"
+	"strings"
+
+	"helm.sh/helm/v4/pkg/plugin/schema"
+)
+
+func getProtocolDownloader(downloaders []SubprocessDownloaders, protocol string) *SubprocessDownloaders {
+	for _, d := range downloaders {
+		if slices.Contains(d.Protocols, protocol) {
+			return &d
```

Comment:
see above. we really just need the command here, and there should only be one. may be good idea to error if more than one command is found for a protocol.

```suggestion
func getProtocolCommand(getterCommands []SubprocessGetterCommands, protocol string) string{
	for _, d := range getterCommands {
		if slices.Contains(d.Protocols, protocol) {
			return d.Command
```